src/main/java/uk/gov/ida/eventemitter/EventEmitterModule.java

```diff
@@ -33,122 +33,107 @@ protected void configure() {}
-                    if (hasAmazonCredentialWithARegion(credentials, configuration)) {
-                        return AmazonS3ClientBuilder.standard()
-                                                    .withCredentials(new AWSStaticCredentialsProvider(credentials.get()))
-                                                    .withRegion(config.getRegion())
-                                                    .build();
-                    }
-                    AmazonS3ClientBuilder.defaultClient();
-                }
-                return null;
-            });
+    @Nullable
+    @Named("SourceQueueUrl")
+    private String getQueueUrl(
+            @Nullable final AmazonSQS amazonSqs,
+            final Configuration configuration) {
+        if (configuration.isEnabled()) {
+            return amazonSqs.getQueueUrl(configuration.getSourceQueueName()).getQueueUrl();
+        }
+        return null;
     }
 
     @Provides
     @Singleton
-    private Optional<AWSKMS> getAmazonKms(
-        final Optional<AmazonS3> amazonS3,
-        final Optional<AWSCredentials> credentials,
-        final Optional<Configuration> configuration) {
-
-        return amazonS3.map(s3 -> {
-            if (hasAmazonCredentialWithARegion(credentials, configuration)) {
-                return AWSKMSClientBuilder.standard()
-                                          .withCredentials(new AWSStaticCredentialsProvider(credentials.get()))
-                                          .withRegion(configuration.get().getRegion())
-                                          .build();
-            }
-            return AWSKMSClientBuilder.defaultClient();
-        });
+    private SqsClient getAmazonSqsClient(
+            @Nullable final AmazonSQS amazonSqs,
+            @Nullable final @Named("SourceQueueUrl") String sourceQueueUrl,
+            final Configuration configuration) {
+        if (configuration.isEnabled()) {
+            return new AmazonSqsClient(amazonSqs, sourceQueueUrl);
+        }
+        return new StubSqsClient();
     }
 
     @Provides
     @Singleton
     private Encrypter getEncrypter(
-        final Optional<AmazonS3> amazonS3,
-        final Optional<Configuration> configuration,
-        final ObjectMapper mapper,
-        final Optional<AWSKMS> amazonKms) {
-
-        if (amazonS3.isPresent() && amazonKms.isPresent()) {
+            @Nullable final AmazonS3 amazonS3,
+            final Configuration configuration,
+            final ObjectMapper mapper,
+            final AWSKMS amazonKms) {
+        if (configuration.isEnabled()) {
             try {
-                S3Object s3Object = amazonS3.get().getObject(configuration.get().getBucketName(), configuration.get().getKeyName());
+                S3Object s3Object = amazonS3.getObject(configuration.getBucketName(), configuration.getKeyName());
                 try (S3ObjectInputStream s3ObjectInputStream = s3Object.getObjectContent()) {
```

Comment:
Please remove the code which catches exception. If Event Emitter fails to create an instance of Event Encrypter, it should throw an exception instead of using Stub Encrypter. We do not want to use StubEncrypter in production since stub encrypter will not encrypt event messages (i.e. there will be loss of event messages). Alternatively, we can rid of all stubs classes since this PR allows event emitter client applications to enable or disable event emitter.

Reply:
Okey dokey. We have made it throw a specific exception now :)

Reply:
Thanks. 👍